src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -3013,9 +3013,14 @@ bool QgsPostgresProvider::changeAttributeValues( const QgsChangedAttributesMap &
         if ( result.PQresultStatus() != PGRES_COMMAND_OK && result.PQresultStatus() != PGRES_TUPLES_OK )
           throw PGException( result );
       }
+      else // let the user know that no field was actually changed
+      {
+        QgsLogger::warning( tr( "No fields were updated on the database." ) );
+      }
 
       // update feature id map if key was changed
-      if ( pkChanged && mPrimaryKeyType == PktFidMap )
+      // PktInt64 also uses a fid map even if it is a stand alone field.
```

Comment:
Maybe, for clarity, you could add a boolean member like `mUseFidMap` or `mFidMapEnabled` to store this information, it would be easier than remembering the exceptional behavior of `PkInt64`.

Reply:
I was going to implement it, but it was merged before I got around it :smile: Yes, it may be useful to have such a flag, because other data types might be handled the same way (I'm looking at real and double PostgreSQL types), but I will leave it for later.